openpype/hosts/houdini/api/lib.py

```diff
@@ -649,3 +655,71 @@ def get_color_management_preferences():
         "display": hou.Color.ocio_defaultDisplay(),
         "view": hou.Color.ocio_defaultView()
     }
+
+
+def publisher_show_and_publish():
+    """Open publisher window and trigger publishing action."""
+
+    main_window = get_main_window()
+    publisher_window = get_tool_by_name(
+        tool_name="publisher",
+        parent=main_window
+    )
+
+    publisher_window.set_current_tab("publish")
+    publisher_window.make_sure_is_visible()
+    publisher_window._reset_on_show = False
```

Comment:
I'm strongly against any usage of private functions or attributes. They're private, so they can be changed in future without checking if are used.

Reply:
Fair enough! 
I will try finding another way.

Reply:
I can add the option if you tell me what you want to do...

Reply:
I was coping from this [code snippet](https://discord.com/channels/517362899170230292/517382145552154634/1151589724914454679) on discord. 

how should we trigger the publish action ?

Reply:
Without that variable set @fabiaserra said the publish would still be interrupted by the UI because it resets on show ;)

Reply:
yeah I'm also against accessing private variables, I just did that as a quick hack